src/app-exception-filters/main-exception.filter.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ExceptionFilter,
+  Catch,
+  ArgumentsHost,
+  HttpException,
+} from '@nestjs/common';
+import { Request, Response } from 'express';
+
+@Catch()
+export class MainExceptionFilter implements ExceptionFilter {
+  catch(exception: Error, host: ArgumentsHost) {
+    const ctx = host.switchToHttp();
+    const response = ctx.getResponse<Response>();
+    const request = ctx.getRequest<Request>();
+    const status =
+      exception instanceof HttpException ? exception.getStatus() : 500;
+
+    this.printErrorToConsole(exception, status);
+
+    response.status(status).json({
```

Comment:
Może warto zrobić jakąś bardziej custom-ową odpowiedź. Na zasadzie, że wyświetli status błędu i jakąś wiadomość otrzymaną z miejsca gdzie jest rzucany błąd?
Wszystkie HTTP Exceptions by noramlnie rzucało, a jeżeli by było coś innego to 500 - Internal Server Error - Sorry for the issue... albo coś podobnego.

Reply:
Wyrzuca do usera jsona z routem, time oraz nr błędu. Szczegóły dla developera na razie są drukowane w konsoli + z czasem podepnie się tam system logów. Z czasem też będzie można to rozwinąć, nie wiem czy jest sens teraz... Ale jak chcecie jak coś :)

Reply:
Napisałem tak na podstawie moich wcześniejszych projektów 🙂 Jak ja to robiłem. Wiadomo, że można to i później zmienić

Reply:
do usera nie ma sensu tego robic bo go to kompletnie nie interesuje - chyba ze jest to jakis konrketny blad np nie ma zasobu z podanym ID to ma sens wtedy. Dla deva jak najbardziej widzialbym jakiegos customowego wiekszego loga